actionview/lib/action_view/railtie.rb

```diff
@@ -48,6 +48,14 @@ class Railtie < Rails::Railtie # :nodoc:
       end
     end
 
+    initializer "action_view.eager_load_templates" do |app|
+      ActiveSupport.on_load(:action_controller) do
+        if Rails.env.production? && app.config.action_view.eager_load_templates
```

Comment:
Why hardcode to `Rails.env.production?` By default the `eager_load_templates` config is only enabled on production, but if developers want to turn it on other envs they should be able to.